src/metrics/timingMetric.js

```diff
@@ -0,0 +1,43 @@
+import {VALID_TYPES} from './metrics';
+
+function typeOrError(type) {
+  if (VALID_TYPES.includes(type)) {
+    return type;
+  }
+  throw new Error('invalid type for a metric supplied');
+}
+
+export default function newMetric(metricType, {fields, tags} = {
```

Comment:
Apart from the `VALID_TYPES`, the coupling is one way. I think it's nice to keep this separate, it keeps the files small and focussed.